DESIGN.md

```diff
@@ -325,18 +325,99 @@ In the addition of the previous rules, an operator for `Flowable`:
 
 ### Creation
 
-Creation of the various types should be exposed through factory methods that provide safe construction.
+Unlike RxJava 1.x, 2.x base classes are abstract and stateless and generally no longer wrap an `OnSubscribe` callback - this saves allocation in assembly time without limiting the expressiveness. Operator methods and standard factories still live as final on the base classes. 
+
+Since Java still doesn't have extension methods, "adding" more operators can only happen through helper methods such as `lift(C -> C)` and `compose(R -> P)` where `C` is the default consumer type (i.e., `rs.Subscriber`), `R` is the base type (i.e., `Flowable`) and `P` is the base interface (i.e., `rs.Publisher`). As before, the library itself may gain or lose standard operators and/or overloads through the same community process.
+
+In concert, `create(OnSubscribe)` will not be available; standard operators extend the base types directly. The conversion of other RS-based libraries will happen through the `Flowable.wrap()` static method. 
+
+(*The unfortunate effect of `create` in 1.x was the ignorance of the Observable contract and beginner's first choice as an entry point. We can't eliminate this path since `rs.Publisher` is a single method functional interface that can be implemented just as badly.*)
+
+Therefore, new standard factory methods will try to address the common entry point requirements: 
+  - `create(SyncOnSubscribe)` to safe, synchronous generation of signals, one-by-one
+  - `create(AsyncOnSubscribe)` to batch-create signals based on request patterns
+  - `create(Consumer<? super SingleEmitter<T>>)` to relay a single value or error from other reactive sources (i.e., addListener callbacks)
+  -  `create(Consumer<? super FlowEmitter<T>>)` to relay multiple values or error from multi-valued reactive-sources (i.e., button-clicks) while also give flow control options right there (buffer, drop, error, etc.).
+
+The first two `create` methods take an implementation of an interface which provides state and the generator methods:
 
 ```java
-Flowable.create(SyncGenerator generator)
+interface SyncOnSubscribe<T, S> {
```

Comment:
@akarnokd I'm pretty sure that the `SyncGenerator` was supposed to be a renamed `SyncOnSubscribe` (and likewise the Asyc variant). Would you mind renaming these interface declarations to match? If anyone remembers otherwise maybe they can correct my memory @abersnaze @benjchristensen @stevegury